### PR TITLE
Use tox for linting and building docs

### PR DESCRIPTION
**Pull request by Luthaf**
This brings chemiscope in line with other projects in the lab; and will make it nicer to build the docs once #334 is merged (since the new code will require installing chemiscope before trying to build the docs)

### CODE REVIEW

**Comment by Luthaf** [comment]
TODO: 

- [ ] update dev docs if we have any to point to `tox`

File: tox.ini [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    tests
+
+
+[testenv]
+package = external
+package_env = build-chemiscope
+lint_folders = "{toxinidir}/python" "{toxinidir}/setup.py"
+
+
+[testenv:build-chemiscope]
```
Comment (PicoCentauri) [inline tox.ini:14]: I think simply `build` is enough. We are not building different flavors as the the other bigger packages.

```suggestion
[testenv:build]
description = builds the package and checks integrity
```

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+deps =
+    jupyterlab
+
+commands =
+    pip wheel . --no-deps --no-build-isolation --check-build-dependencies --wheel-dir {envtmpdir}/dist
+
+
+[testenv:tests]
```
Comment (PicoCentauri) [inline tox.ini:22]:
```suggestion
[testenv:tests]
description = Run Python unit tests
```

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+deps =
+    ase
+
+commands =
+    python -m unittest discover -p "*.py" -s python/tests
+
+
+[testenv:lint]
```
Comment (PicoCentauri) [inline tox.ini:30]:
```suggestion
[testenv:lint]
description =
    lint the Python code with flake8 (code linter), black (code formatter), and isort
    (sorting of imports)
```

File: tox.ini [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    blackdoc
+    isort
+
+commands =
+    flake8 --extend-exclude "*/build/*" {[testenv]lint_folders}
+    black --check --diff {[testenv]lint_folders}
+    blackdoc --check --diff {[testenv]lint_folders}
+    isort --check-only --diff --extend-skip-glob "*/build/*" {[testenv]lint_folders}
+
+[testenv:format]
```
Comment (PicoCentauri) [inline tox.ini:45]:
```suggestion
[testenv:format]
description = Abuse tox to do actual formatting on all files.
```

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    black
+    blackdoc
+    isort
+commands =
+    black {[testenv]lint_folders}
+    blackdoc {[testenv]lint_folders}
+    isort {[testenv]lint_folders}
+
+
+[testenv:docs]
```
Comment (PicoCentauri) [inline tox.ini:57]:
```suggestion
[testenv:docs]
description = Build the package documentation.
```

**Review by PicoCentauri — COMMENTED** [review]
Maybe use the description section.

File: tox.ini (hunk 1 above)
Comment (Luthaf) [reply tox.ini]: done!

**Comment by PicoCentauri** [comment]
Looks good. Do you want to update the dev docs now or later?

**Comment by Luthaf** [comment]
I've updated the Contributing, that's all the dev docs we have on this repo!

**Review by PicoCentauri — APPROVED** [review]

**Pull request merged**